sqs/src/main/scala/akka/stream/alpakka/sqs/SqsSourceStage.scala

```diff
@@ -4,44 +4,77 @@
 package akka.stream.alpakka.sqs
 
 import java.util
+import java.util.concurrent.Executors
 
 import akka.stream.stage.{GraphStage, GraphStageLogic, OutHandler}
 import akka.stream.{Attributes, Outlet, SourceShape}
+import com.amazonaws.auth.{AWSCredentials, DefaultAWSCredentialsProviderChain}
 import com.amazonaws.handlers.AsyncHandler
 import com.amazonaws.services.sqs.AmazonSQSAsyncClient
 import com.amazonaws.services.sqs.model.{Message, ReceiveMessageRequest, ReceiveMessageResult}
 
 import scala.collection.JavaConverters._
 
 object SqsSourceSettings {
-  val Defaults = SqsSourceSettings(20, 100, 10)
+  val Defaults = SqsSourceSettings(20, 100, 10, None)
+
+  def create(waitTimeSeconds: Int,
+             maxBufferSize: Int,
+             maxBatchSize: Int,
+             credentials: AWSCredentials): SqsSourceSettings =
+    SqsSourceSettings(waitTimeSeconds, maxBufferSize, maxBatchSize, Some(credentials))
+
+  private def create(waitTimeSeconds: Int, maxBufferSize: Int, maxBatchSize: Int): SqsSourceSettings =
+    SqsSourceSettings(waitTimeSeconds, maxBufferSize, maxBatchSize, None)
+
 }
 
 //#SqsSourceSettings
-final case class SqsSourceSettings(waitTimeSeconds: Int, maxBufferSize: Int, maxBatchSize: Int) {
+final case class SqsSourceSettings(
+    waitTimeSeconds: Int,
+    maxBufferSize: Int,
+    maxBatchSize: Int,
+    credentials: Option[AWSCredentials]
+) {
   require(maxBatchSize <= maxBufferSize)
   // SQS requirements
   require(waitTimeSeconds >= 0 && waitTimeSeconds <= 20)
   require(maxBatchSize >= 1 && maxBatchSize <= 10)
 }
 //#SqsSourceSettings
 
-final class SqsSourceStage(queueUrl: String, settings: SqsSourceSettings, sqsClient: AmazonSQSAsyncClient)
-    extends GraphStage[SourceShape[Message]] {
+final class SqsSourceStage(queueUrl: String, settings: SqsSourceSettings) extends GraphStage[SourceShape[Message]] {
```

Comment:
I'll try to introduce another way to initialize the `SqsSource` with a `AmazonSQSAsync` instead of only creating one explicitly. @aserrallerios came up with the idea to initialize the client within the `SqsSourceStage` so the threadpool can be configured properly. Parallelism is now tied to `maxBufferSize` and `maxBatchSize` so the maximum size for the thread pool can be defined with `maxBufferSize / maxBatchSize`.